sparkles/core.py

```diff
@@ -550,10 +550,19 @@ def __init__(self, *args, **kwargs):
         # Compute guide count once for the record
         # TODO make this a property
         if self.guides is not None:
-            self.guide_count = guide_count(self.guides['mag'], self.guides.t_ccd)
+            mags = self.guides['mag']
+            t_ccds = np.full_like(mags, self.guides.t_ccd)
+            if self.dyn_bgd_n_faint > 0:
+                # Apply the dynamic background t_ccd bonus to the
+                # dyn_bgd_n_faint faintest stars. See:
+                # https://nbviewer.org/urls/cxc.harvard.edu/mta/ASPECT/ipynb/misc/guide-count-dyn-bgd.ipynb
+                mags = np.sort(mags)
+                n_faint = min(self.dyn_bgd_n_faint, len(t_ccds))
```

Comment:
this is not needed. The following is perfectly fine:
```python
t_ccds[-self.dyn_bgd_n_faint:] += self.dyn_bgd_dt_ccd
```
you can try it yourself:
```python
In [1]: a = np.arange(6)

In [2]: a[-10:]
Out[2]: array([0, 1, 2, 3, 4, 5])
```

Reply:
I had some recollection of being bit by that, but either something changed (not likely) or I remembered wrong (likely). In any case I'll fix this.